csg.js

```diff
@@ -88,20 +88,31 @@ for solid CAD anyway.
         return a - b;
     }
 
-    // # class CSG
-    // Holds a binary space partition tree representing a 3D solid. Two solids can
-    // be combined using the `union()`, `subtract()`, and `intersect()` methods.
+    /** Class CSG
+     * Holds a binary space partition tree representing a 3D solid. Two solids can
+     * be combined using the `union()`, `subtract()`, and `intersect()` methods.
+     * @constructor
+     */
     var CSG = function() {
         this.polygons = [];
         this.properties = new CSG.Properties();
         this.isCanonicalized = true;
         this.isRetesselated = true;
     };
 
-    CSG.defaultResolution2D = 32;
+    /** Number of polygons per 360 degree revolution for 2D objects.
+     * @default
+     */
+    CSG.defaultResolution2D = 32; // FIXME this seems excessive
+    /** Number of polygons per 360 degree revolution for 3D objects.
+     * @default
+     */
     CSG.defaultResolution3D = 12;
 
-    // Construct a CSG solid from a list of `CSG.Polygon` instances.
+    /** Construct a CSG solid from a list of `CSG.Polygon` instances.
+     * @param {CSG.Polygon[]} polygons - list of polygons
+     * @returns {CSG} new CSG object
```

Comment:
Small question : does JSDOC reckognize the {CSG} object as it is ? or does it need to be '{object}' : haven't tried this before.